File: php/examples/get_resources_download.php
<?php
require_once 'base_request.php';

// Example of GET /resources/:resource_id/download
// The specified resource is saved as a pdf in the 'documents' directory

$debug = true;
$resource_id   = $_ENV['AGENCY_RESOURCE_ID'];
$request_data = '';
$request_params = [];

send_request('GET', '/resources/' . $resource_id . '/download', $request_params, $request_data);

File: csharp/examples/GetResourcesDownload.cs
using System;

namespace Examples.examples
{
    // Example of GET /resources/:resource_id/download

    // The specified resource is saved as a pdf in the 'documents' directory
    
    class GetResourcesDownload : BaseRequest
    {
        private string _resourceId;

        public GetResourcesDownload() : base()
        {
            dotenv.net.DotEnv.Load();
            _resourceId = Environment.GetEnvironmentVariable("AGENCY_RESOURCE_ID");
            Console.WriteLine($"GetResourcesDownload: resource_id={_resourceId}");
        }

        public override string SendRequest()
        {
            string path = $"resources/{_resourceId}/download";
            return GetRequest(path);
        }
    }
}
